sklearn/feature_extraction/tests/test_feature_hasher.py

```diff
@@ -107,3 +107,21 @@ def test_hasher_zeros():
     # Assert that no zeros are materialized in the output.
     X = FeatureHasher().transform([{'foo': 0}])
     assert_equal(X.data.shape, (0,))
+
+
+def test_hasher_non_negative():
+    raw_X = [["foo", "bar", "baz"]]
+
+    def it():  # iterable
+        return (x for x in raw_X)
+
+    X = FeatureHasher(non_negative=False,
+                      input_type='string').fit_transform(it())
+    assert_true((X.data > 0).any() and (X.data < 0).any())
```

Comment:
Well the point here was to test that the output has both negative and positive values, not that is has zeros (which it doesn't in this case). Changed the test to be more verbose about it.